experimental/dds/tree2/src/simple-tree/toMapTree.ts

```diff
@@ -161,13 +164,44 @@ function valueToMapTree(
 	const schema = getSchema(context, type);
 	assert(allowsValue(schema.leafValue, value), "Unsupported schema for provided primitive.");
 
+	const mappedValue = mapUnsupportedPrimitive(value, typeSet);
+
 	return {
-		value,
+		value: mappedValue,
 		type,
 		fields: new Map(),
 	};
 }
 
+/**
+ * Checks an incoming value to ensure it is compatible with our serialization format.
+ * For unsupported values with a schema-compatible replacement, return the replacement value.
+ * For unsupported values without a schema-compatible replacement, throw.
+ * For supported values, return the input.
+ */
+function mapUnsupportedPrimitive(
+	// eslint-disable-next-line @rushstack/no-new-null
+	value: boolean | number | string | IFluidHandle | null,
+	typeSet: TreeTypeSet,
+	// eslint-disable-next-line @rushstack/no-new-null
+): boolean | number | string | IFluidHandle | null {
+	if (Object.is(value, -0)) {
+		// Our serialized data format does not support -0.
+		// Map such input to +0.
+		return 0;
+	} else if (Number.isNaN(value) || !Number.isFinite(value)) {
+		// Our serialized data format does not support NaN nor +/-∞.
+		// If the schema supports `null`, fall back to that. Otherwise, throw.
+		if (typeSet?.has(leaf.null.name) ?? false) {
+			return null;
+		} else {
+			throw new TypeError(`Received unsupported numeric value: ${value}.`);
```

Comment:
I'm confused as to why we don't hit this code path all the time for non-numbers. If I pass in `"hello"`, then it's not `-0`, it's not a finite number, and if my schema doesn't support `null`, won't I throw this error instead of just returning `"hello"`?

I.e. this supports numbers but not the other primitives. Are you planning on adding the support for the other types soon?

Reply:
Haha, this was an oversight that I hadn't yet caught because I left an `only` on my new tests... Fixed now.